pkg/live/inventoryrg.go

```diff
@@ -228,20 +230,71 @@ var crdGroupKind = schema.GroupKind{
 	Kind:  "CustomResourceDefinition",
 }
 
-// ResourceGroupCRDApplied returns true if the inventory ResourceGroup
-// CRD is available from the current RESTMapper, or false otherwise.
-func ResourceGroupCRDApplied(factory cmdutil.Factory) bool {
+// ResourceGroupCRDApplied returns (true, true) if the inventory ResourceGroup
+// CRD is available from the current RESTMapper and matches the latest declaration.
+// It returns (false, false) if the inventory ResourceGroup CRD is not installed.
+// It returns (true, false) if the ResourceGroup CRD is installed, but
+// doesn't match the latest declaration.
+func ResourceGroupCRDApplied(factory cmdutil.Factory) (bool, bool) {
 	mapper, err := factory.ToRESTMapper()
 	if err != nil {
 		klog.V(4).Infof("error retrieving RESTMapper when checking ResourceGroup CRD: %s\n", err)
-		return false
+		return false, false
 	}
 	_, err = mapper.RESTMapping(ResourceGroupGVK.GroupKind())
 	if err != nil {
 		klog.V(7).Infof("error retrieving ResourceGroup RESTMapping: %s\n", err)
+		return false, false
+	}
+	return true, resourceGroupCRDMatchesLatest(mapper, factory)
+}
+
+func resourceGroupCRDMatchesLatest(mapper meta.RESTMapper, factory cmdutil.Factory) bool {
+	rgCRDStr, ok := resourceGroupCRDs["v1"]
```

Comment:
I think we can probably ignore the error check here. We know it is there and the code will panic if for some reason it is not.

Reply:
Do you mean that we don't check if the ResourceGroup CRD in the cluster is the same as the one in the current kpt release? If we do that, users may see some error from the apply library, but it doesn't provide the meaning action to fix the error.

Reply:
Please ignore my previous comments. I see what you mean. Let me update it.

Reply:
done